### ISSUE TITLE
Translate rotated x-axis labels on bar charts

### ISSUE THREAD

**Issue by Tchanders** [mozilla/metrics-graphics#457]
We're trying to re-draw the graphs on https://crash-stats.mozilla.com using MetricsGraphics. I'm currently having trouble rotating our (rather long) x-axis labels on a bar chart. Un-rotated they look like this:
![screenshot from 2015-07-08 21 03 03](https://cloud.githubusercontent.com/assets/10440647/8581030/fe688ff6-25b5-11e5-8048-88d9844bad60.png)

Rotated 90 degrees anti-clockwise (rotate_x_labels: -90):
![screenshot from 2015-07-08 21 03 58](https://cloud.githubusercontent.com/assets/10440647/8581037/04311a3e-25b6-11e5-8367-650210ec57c7.png)

Rotated 90 degrees clockwise (rotate_x_labels: 90):
![screenshot from 2015-07-08 21 02 05](https://cloud.githubusercontent.com/assets/10440647/8581038/074f7cc4-25b6-11e5-931c-2ff8688d2fe5.png)

If the svg has a fixed height then perhaps the labels should be translated as well as rotated, and the rest of the graph resized accordingly?

**Comment by almossawi** [comment]
You'll want to modify [bottom](https://github.com/mozilla/metrics-graphics/wiki/Layout#bottom) to get the labels to appear. Yes, we probably ought to have reasonable defaults in cases where labels get clipped.

Bar charts are still a work-in-progress, so feel free to file bugs for any and all issues that you come across.

**Comment by Tchanders** [comment]
Thanks, modifying bottom sorts out the anticlockwise situation, but the labels still collide when rotating clockwise. Have suggested a fix by changing the text-anchor property depending on the direction of the rotation.

I've also moved rotate after truncate since it might help to know the actual width of the text in order to centre the labels properly in the future (the first approach I tried depended on this).
